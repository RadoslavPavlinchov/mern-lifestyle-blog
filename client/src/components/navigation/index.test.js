import React from 'react';
import styles from './index.module.css'
import renderer from 'react-test-renderer';
import Navigation from '.';
import UserContext from '../../Context';
import { BrowserRouter, Route, Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import socialLinks from '../../utils/social-links';

describe('Navigation Component', () => {
    it('should render authenticated routes', () => {
        const socLinks = socialLinks();
        const tree = renderer.create(
            <BrowserRouter>
                <Route>
                    <UserContext.Provider value={{
                        user: {
                            loggedIn: true,
                            id: '123456789'
                        }
                    }}>
                        <Navigation>
                            {
                                socLinks.map((link, index) => (
                                    <Link to='/'
                                        key={index}
                                    >
                                        <FontAwesomeIcon icon={[link.type, link.code]} className={styles.navIcon} />
                                    </Link>
                                ))
                            }
                        </Navigation>
                    </UserContext.Provider>
                </Route>
            </BrowserRouter>

        ).toJSON();
        expect(tree).toMatchSnapshot()
    })

    // it('should render authenticated routes', () => {

    // })
})