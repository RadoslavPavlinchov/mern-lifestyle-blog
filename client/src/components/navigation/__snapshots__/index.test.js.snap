// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Component should render authenticated routes 1`] = `
<nav
  className="nav"
>
  <div
    className="navMenu"
  >
    <div>
      <a
        className="brand"
        href="/"
        onClick={[Function]}
      >
        Lifestyle
      </a>
    </div>
    <div>
      <ul
        className="navItems"
      >
        <li
          className="navLink"
        >
          <a
            className="navAnchor"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="navLink"
        >
          <a
            className="navAnchor"
            href="/categories"
            onClick={[Function]}
          >
            Categories
          </a>
        </li>
        <li
          className="navLink"
        >
          <a
            className="navAnchor"
            href="/contact-us"
            onClick={[Function]}
          >
            About Us
          </a>
        </li>
        <li
          className="navLink"
        >
          <a
            className="navAnchor"
            href="/profile/123456789"
            onClick={[Function]}
          >
            Profile
          </a>
        </li>
        <li
          className="navLink"
        >
          <a
            className="navAnchor"
            href="/admin"
            onClick={[Function]}
          >
            Admin
          </a>
        </li>
      </ul>
    </div>
    <div
      className="social"
    >
      <a
        href="/"
        onClick={[Function]}
      />
      <a
        href="/"
        onClick={[Function]}
      />
      <a
        href="/"
        onClick={[Function]}
      />
      <a
        href="/"
        onClick={[Function]}
      />
    </div>
  </div>
</nav>
`;
